### PR TITLE
Make TemplateFile parameter optional when used with .bicepparam parameter file

### PR DESCRIPTION
**Pull request by anthony-c-martin**
## Description

Support for .bicepparam was introduced with #21567.

In order to support .bicepparam files which link directly to Bicep Registry, Template Spec or JSON content, we must relax the requirement for the user to provide the -TemplateFile parameter for deployment-related commands. This parameter is currently redundant, as .bicepparam files require the user to link to a .bicep file via the using statement.

To maintain backwards-compatibility, continue to support the "old" command if the user supplies the TemplateFile parameter.

### Example
```powershell
New-AzResourceGroupDeployment `
  -ResourceGroup my-rg `
  -TemplateParameterFile ./params.bicepparam
```

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Resources/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 -->
 
 ## Upcoming Release
+* Add ability to deploy directly from a `.bicepparam` file.
```
Comment (BethanyZhou) [inline src/Resources/Resources/ChangeLog.md:22]:
```suggestion
* Supported `TemplateParameterFile ` to accept a `.bicepparam` file.
```

File: src/Resources/Resources/help/Set-AzManagementGroupDeploymentStack.md [hunk 2]
```diff
@@ -141,6 +141,16 @@ Set-AzManagementGroupDeploymentStack -Name <String> -ManagementGroupId <String>
  [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
 ```
 
+### ByParameterFileWithNoTemplate
+```
+Set-AzManagementGroupDeploymentStack -Name <String> -ManagementGroupId <String>
+ -DeploymentSubscriptionId <String> -Location <String> [-Description <String>] [-DeleteAll] [-DeleteResources]
```
Comment (BethanyZhou) [inline src/Resources/Resources/help/Set-AzManagementGroupDeploymentStack.md:147]: Please add an example for each cmdlet at least if one parameter set is added.

File: src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs [hunk 3]
```diff
@@ -79,6 +85,8 @@ protected DeploymentStacksCreateCmdletBase()
             Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Parameter file to use for the template.")]
         [Parameter(ParameterSetName = ParameterFileTemplateSpecParameterSetName,
             Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Parameter file to use for the template.")]
+        [Parameter(ParameterSetName = ByParameterFileWithNoTemplateParameterSetName,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Parameter file to use for the deployment.")]
```
Comment (BethanyZhou) [inline src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs:89]: Only help message in the first parameter set will be considered in help MDs. Please change to a conclusive HelpMessage.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Resources/Resources/help/Set-AzManagementGroupDeploymentStack.md (hunk 2 above)
Comment (anthony-c-martin) [reply src/Resources/Resources/help/Set-AzManagementGroupDeploymentStack.md:147]: Done!

File: src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs (hunk 3 above)
Comment (anthony-c-martin) [reply src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs]: Done!

File: src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentCmdletBase.cs [hunk 4]
```diff
@@ -84,13 +90,15 @@ protected ResourceWithParameterCmdletBase()
         public Hashtable TemplateParameterObject { get; set; }
 
         [Parameter(ParameterSetName = TemplateObjectParameterFileParameterSetName,
-            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "A file that has the template parameters.")]
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Parameter file to use for the template.")]
         [Parameter(ParameterSetName = TemplateFileParameterFileParameterSetName,
-            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "A file that has the template parameters.")]
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Parameter file to use for the template.")]
```
Comment (BethanyZhou) [inline src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentCmdletBase.cs:95]: Please update parameter description in MDs as well

Comment (anthony-c-martin) [reply src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentCmdletBase.cs:95]: 👍

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**